### PR TITLE
Supported mysql BIT data type.

### PR DESCRIPTION
**Pull request by Lucas-307**
Fixes part of #4795.
Supported mysql BIT data type.

### CODE REVIEW

File: shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/integer/MySQLBitBinlogProtocolValue.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.apache.shardingsphere.db.protocol.mysql.packet.binlog.row.column.MySQLBinlogColumnDef;
+import org.apache.shardingsphere.db.protocol.mysql.packet.binlog.row.column.value.MySQLBinlogProtocolValue;
+import org.apache.shardingsphere.db.protocol.mysql.payload.MySQLPacketPayload;
+
+import java.io.Serializable;
+
+/**
+ * BIT type value of MySQL binlog protocol.
+ */
+public class MySQLBitBinlogProtocolValue implements MySQLBinlogProtocolValue {
```
Comment (terrymanu) [inline shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/integer/MySQLBitBinlogProtocolValue.java:29]: It is better to add final if the class is not for extended

File: shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/integer/MySQLBitBinlogProtocolValue.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    @Override
+    public Serializable read(final MySQLBinlogColumnDef columnDef, final MySQLPacketPayload payload) {
+        return payload.readLong(getLength(columnDef.getColumnMeta()));
+    }
+    
+    private int getLength(final int meta) {
+        int nbits = ((meta >> 8) * 8) + (meta & 0xff);
+        return (nbits + 7) / 8;
+    }
+}
```
Comment (terrymanu) [inline shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/integer/MySQLBitBinlogProtocolValue.java:40]: Can we consider about reuse with MySQLNullBitmap?

File: shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/payload/MySQLPacketPayload.java [hunk 3]
```diff
@@ -222,6 +222,19 @@ public void writeIntLenenc(final long value) {
         byteBuf.writeLongLE(value);
     }
     
+    /**
+     * Read fixed length long from byte buffers.
+     * @param length length read from byte buffers
```
Comment (terrymanu) [inline shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/payload/MySQLPacketPayload.java:227]: Please leave a blank line between java doc tile and `@param`

File: shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/test/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/integer/MySQLBitBinlogProtocolValueTest.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MySQLBitBinlogProtocolValueTest {
```
Comment (terrymanu) [inline shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/test/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/integer/MySQLBitBinlogProtocolValueTest.java:33]: It is better to add final if the class is not for extended

File: shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/integer/MySQLBitBinlogProtocolValue.java (hunk 2 above)
Comment (tuohai666) [reply shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/integer/MySQLBitBinlogProtocolValue.java:40]: I have discuss with Lucas. I think we can't resule MySQLNullBitmap, they are different.

**Pull request merged**